src/hdf5_dl.c

```diff
@@ -97,44 +105,209 @@ hid_t H5E_ERR_CLS_g = -1;
 
 static bool is_init = false;
 
+/*
+ * Try to find a symbol within a library
+ * 
+ * handle: Handle to the library
+ * symbol: Symbol to look for
+ * Returns: a pointer to the symbol or NULL 
+ * if the symbol can't be found 
+ */
+void *find_sym(void *handle, const char *symbol) {
+
+  void *ret = NULL, *err = NULL;
+  dlerror(); /* clear error code */
+  ret = dlsym(handle, symbol);
+
+  if(ret != NULL && (err = dlerror()) == NULL)
+    return ret;
+  else
+    return NULL;
+}
+
+/*
+ * Check that all symbols have been loaded
+ * 
+ * Returns: 0 if an error occured, -1 for success
+ */
+int check_symbols() {
+
+  int retval = -1;
+
+  if(DL_H5Functions.H5open == NULL)
+    retval = 0;
+
+  /*H5E*/
+  if(DL_H5Functions.H5Epush1 == NULL)
+    retval = 0;
+
+  if(DL_H5Functions.H5Epush2 == NULL)
+    retval = 0;
+
+  /*H5P*/
+  if(DL_H5Functions.H5Pget_filter_by_id2 == NULL)
+    retval = 0;
+
+  if(DL_H5Functions.H5Pget_chunk == NULL)
+    retval = 0;
+
+  if(DL_H5Functions.H5Pmodify_filter == NULL)
+    retval = 0;
+
+  /*H5T*/
+  if(DL_H5Functions.H5Tget_size == NULL)
+    retval = 0;
+
+  if(DL_H5Functions.H5Tget_class == NULL)
+    retval = 0;
+  
+  if(DL_H5Functions.H5Tget_super == NULL)
+    retval = 0;
+  
+  if(DL_H5Functions.H5Tclose == NULL)
+    retval = 0;
+
+  /*H5Z*/
+  if(DL_H5Functions.H5Zregister == NULL)
+    retval = 0;
+
+  /*Variables*/
+  if(H5E_CANTREGISTER_g == -1)
```

Comment:
This is the same problem as before: -1 is a value that might have been assigned to this value by libhdf5.  You need to check whether the pointer is NULL or not.